src/servers/Server.cc

```diff
@@ -85,16 +85,25 @@ Server::maybeMakeSpaceAvailable()
         debugs(33, 4, "request buffer full: client_request_buffer_max_size=" << Config.maxRequestBufferSize);
```

Comment:
Can we remove this check because the Assure() below guarantees that `inBuf` has some space?

Reply:
Yes, from the compiler point of view, we can, but I do not think we should do that in this PR: A function should not assume or rely on independent checks/assurances in its caller. For now, maybeMakeSpaceAvailable() itself does _not_ assure that inBuf has or gains space; without other changes, we cannot safely move that Assure() from the caller into this method, for example, because future code may call maybeMakeSpaceAvailable() without checking mayBufferMoreRequestBytes() first.

FWIW, I did consider refactoring maybeMakeSpaceAvailable() to guarantee space, but rejected that idea because it would obscure primary PR changes and probably make related code more complex. If we find a good way to refactor, we should do that in a dedicated PR IMO.